lib/client/api.go

```diff
@@ -2474,13 +2478,27 @@ func (tc *TeleportClient) Login(ctx context.Context) (*Key, error) {
 
 	var response *auth.SSHLoginResponse
 
-	switch pr.Auth.Type {
-	case constants.Local:
+	switch authType := pr.Auth.Type; {
+	case pwdless: // Takes precedence over other methods if set.
+		// Do a few sanity checks before obeying.
+		switch {
+		case authType != constants.Local:
+			return nil, trace.BadParameter("Passwordless is only available for local authentication")
```

Comment:
The fact that `pr, err := tc.PingAndShowMOTD(ctx)` returned non-local auth means that local isn't the default, not that it isn't enabled at all. We can check that fact by pinging `/webapi/ping/local` (this will happen if `tc.AuthConnector == "local"`). I think it would make sense for Teleport to use e.g. SAML + passwordless. Initial login would go via SAML, but on regular basis passwordless would be used. 

This would inhibit attribute flow from SAML, so we may need a feature down the line that forces non-passwordless login every X hours/days.

Reply:
My point was that currently if we have say `{saml, local}` as allowed login methods, and `saml` is the default (which is a sensible default to have), then `tsh login --pwdless` will always fail. On the other hand `tsh login --pwdless --auth=local` may work.

1. Should `--pwdless` imply `--auth=local`?
2. Do we *want* to fail entire login flow if passwordless path cannot proceed?

Reply:
In short:

1) It kind of does (see below).
2) I think that's alright (see below too).

There is a certain level of trust behind the --pwdless flag - we trust that they have a proper key registered and available, because we can't check for it beforehand. With that in mind, I think we are OK with the current saml/oidc interaction.

That said, we had a sync yesterday and most of this behavior is going to change. We want to do away with the explicit --pwdless flag and rely on cluster settings / --auth. Still I'd like to get this in as-is and just refactor in the near future.